### PR TITLE
ARIA-408 remove execution creation from workflow runner

### PR DESCRIPTION
**Pull request by mxmrlv**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by aviyoop** [comment]
Add a commit message explain why this change was done.

File: tests/orchestrator/execution/test_execution_compiler.py [hunk 1]
```diff
@@ -296,171 +230,161 @@ def _setup_mock_workflow_in_service(request, inputs=None):
...
+        node = ctx.model.node.refresh(node)
 
-        tasks = workflow_context.model.task.list(filters={'_stub_type': None})
+        tasks = ctx.model.task.list(filters={'_stub_type': None})
         assert any(task.status == task.SUCCESS for task in tasks)
         assert any(task.status == task.RETRYING for task in tasks)
         custom_events['is_resumed'].set()
         assert any(task.status == task.RETRYING for task in tasks)
 
         # Create a new workflow runner, with an existing execution id. This would cause
```
Comment (aviyoop) [inline tests/orchestrator/execution/test_execution_compiler.py]: This comment needs to be revised

File: aria/orchestrator/execution_preparer.py [hunk 2]
```diff
@@ -31,70 +31,79 @@
...
+    """
+    This class manages any execution and tasks related preparation for an execution of a workflow.
+    """
     def __init__(
             self,
-            model,
-            resource,
-            plugin,
+            model_storage,
+            resource_storagee,
```
Comment (aviyoop) [inline aria/orchestrator/execution_preparer.py]: storage

File: aria/orchestrator/execution_preparer.py [hunk 3]
```diff
@@ -31,70 +31,79 @@
...
-                service_id=self._execution.service.id,
-                execution_id=self._execution.id,
-                workflow_name=self._execution.workflow_name,
-                task_max_attempts=self._task_max_attempts,
-                task_retry_interval=self._task_retry_interval,
-            )
-        return self._workflow_context
-
-    def compile(self, execution_inputs=None, executor=None, execution_id=None):
+    def get_ctx(self, execution):
```
Comment (aviyoop) [inline aria/orchestrator/execution_preparer.py]: get_workflow_ctx

File: aria/orchestrator/execution_preparer.py [hunk 4]
```diff
@@ -31,70 +31,79 @@
...
+            service_id=execution.service.id,
+            execution_id=execution.id,
+            workflow_name=execution.workflow_name,
+            task_max_attempts=self._task_max_attempts,
+            task_retry_interval=self._task_retry_interval,
+        )
+
+    def prepare(self, execution_inputs=None, executor=None, execution_id=None):
+        """
+        Prepares the execution and return the workflow ctx. If the execution is new, an execution
```
Comment (aviyoop) [inline aria/orchestrator/execution_preparer.py]: rephrase. The first sentence does not add anything anyway.

File: aria/orchestrator/execution_preparer.py [hunk 5]
```diff
@@ -31,70 +31,79 @@
...
+        and tasks models would be created. A workflow context for the appropriate execution would
+        be created.
+
+        :param execution_inputs: Inputs for the execution.
```
Comment (aviyoop) [inline aria/orchestrator/execution_preparer.py]: Inconsistency with capitalization

**Pull request merged**